src/sentry/static/sentry/app/views/discover/index.tsx

```diff
@@ -87,6 +74,18 @@ class DiscoverContainer extends React.Component<Props, State> {
     this.queryBuilder = createQueryBuilder(query, organization);
   }
 
+  static getDerivedStateFromProps(nextProps: Props, currState): State {
```

Comment:
Do we *need* getDerivedStateFromProps here? It seems like a lot of this can be achieved within componentDidUpdate block. It's likely i'm not up to date with the latest in React but I find it rather confusing to follow state changes.

Reply:
Don't really need it but React docs recommends updating the state with new props by using `getDerivedStateFromProps` and side-effects to be fired by `componentDidUpdate`. 

It seems to be the "best practice" to separate state changes within the component from side-effects outside the components.

Reply:
And because `getDerivedStateFromProps` is a static method, you'll get the `nextProps`+`prevState` but you cannot access `prevProps` and the docs recommends putting the specific values into state if it is needed to get `nextState`.

🤷‍♂️

Reply:
From React docs:
This method exists for rare use cases where the state depends on changes in props over time. For example, it might be handy for implementing a <Transition> component that compares its previous and next children to decide which of them to animate in and out.

It doesn't seem like we are accessing the prev props or care about props change over time here?

It might be better question for TSC / FE team anyway